ner_v1/detectors/base_detector.py

```diff
@@ -13,21 +13,35 @@ class BaseDetector(object):
     detectors using translation
     
     Attributes:
-        supported_languages (set): Set of languages supported by a particular detector
-        source_language_script (str): ISO-639 language code of query text for detection
         target_language_script (str): ISO-639 language code in which detector would process the query
         translation_enabled (bool): Decides to either enable or disable translation API
     """
     __metaclass__ = abc.ABCMeta
 
-    supported_languages = set()
-    source_language_script = ENGLISH_LANG
     target_language_script = ENGLISH_LANG
     translation_enabled = False
 
     def __init__(self):
         pass
 
+    @abc.abstractproperty
+    def supported_languages(self):
+        """
+        This method returns the list of languages supported by entity detectors        
+        Return:
+             list: List of ISO 639 codes of languages supported by subclass/detector
+        """
+        return []
+
+    @abc.abstractproperty
+    def source_language_script(self):
```

Comment:
This shouldn't be a compulsory property for inheriting classes. This can be a Base's attribute and an argument to Base's init. The way I see it Base should only provide supported languages

```python
class BaseDetector(object):
    def __init__(self, source_language_script=ENGLISH_LANG, translation_enabled=False):
        self._source_language_script = source_language_script
        self._target_language_script = ENGLISH_LANG # default, it is decided later on in _set_language_processing_script
        self._translation_enabled = translation_enabled
        self._set_language_processing_script()
        
    def set_translation(enabled):
        self._translation_enabled = enabled
        self._set_language_processing_script()
...
class TextDetector(BaseDetector):
    def __init__(self, entity_name=None, source_language_script=ENGLISH_LANG):
        # Initialize everything needed
        # last step, setup language stuff in base
        super(TextDetector, self).__init__(source_language_script=source_language_script)
        # override if anything needed
```

Reply:
Done